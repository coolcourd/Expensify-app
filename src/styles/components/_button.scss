.button{
  color: white;
  border: none;
  border-radius: 2px;
  background-color: $blue;
  padding: $s-size;
  font-weight: 500;
  font-size: $font-size-large;
  display: inline-block;
  text-decoration: none;
  line-height: 1;
}

.button--link{
  background: none;
}

.button--secondary {
  background: #888;
}
