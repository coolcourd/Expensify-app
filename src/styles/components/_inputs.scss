.text-input {
  border: 1px solid #cacccd;
  height: 50px;
  padding: $s-size;
  font-size: $font-size-large;
  font-weight: 300;
  border-radius: 4px;
}

.select {
  @extend .text-input;
  background-color: $off-white;
}

.textarea {
  @extend .text-input;
  height: 10rem;
}
