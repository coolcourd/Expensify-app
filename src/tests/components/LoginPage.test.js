import React from 'react';
import { shallow } from 'enzyme';
import { Login } from '../../components/LoginPage';

test('Should render login page correctly', () => {
const wrapper = shallow(<Login />);
expect(wrapper).toMatchSnapshot();
});

test('Should call startLogin on button click', () => {
  const startLogin = jest.fn();
  const wrapper = shallow(<Login startLogin={startLogin}/>);
  wrapper.find('button').simulate('click');
  expect(startLogin).toHaveBeenCalled();
});
