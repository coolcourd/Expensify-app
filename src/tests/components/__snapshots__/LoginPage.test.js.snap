// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render login page correctly 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <h1
      className="box-layout__title"
    >
      Expensify
    </h1>
    <p>
      Because you should do it already.
    </p>
    <button
      className="button"
    >
      Login With Google
    </button>
  </div>
</div>
`;
